'test_scale_bigtable' flakes on AssertionError

From [this Kokoro build](https://source.cloud.google.com/results/invocations/5535c655-b8a9-4363-855d-9abc264b6e02/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-bigtable%2Fsamples%2Fpython3.8%2Fpresubmit/log):

```python
_____________________________ test_scale_bigtable ______________________________

instance = <google.cloud.bigtable.instance.Instance object at 0x7f3490cb2490>

    def test_scale_bigtable(instance):
        bigtable_client = bigtable.Client(admin=True)

        instance = bigtable_client.instance(BIGTABLE_INSTANCE)
        instance.reload()

        cluster = instance.cluster(BIGTABLE_INSTANCE)
        cluster.reload()
        original_node_count = cluster.serve_nodes

        scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, True)

        for n in range(10):
            time.sleep(10)
            cluster.reload()
            new_node_count = cluster.serve_nodes
            try:
>               assert (new_node_count == (original_node_count + SIZE_CHANGE_STEP))
E               assert 1 == (1 + 3)

metricscaler_test.py:122: AssertionError
----------------------------- Captured stderr call -----------------------------
Scaled up from 1 to 4 nodes.
------------------------------ Captured log call -------------------------------
INFO     bigtable.metricscaler:metricscaler.py:121 Scaled up from 1 to 4 nodes.
```

--- a/samples/metricscaler/metricscaler_test.py
+++ b/samples/metricscaler/metricscaler_test.py
@@ -15,7 +15,6 @@
 """Unit and system tests for metricscaler.py"""
 
 import os
-import time
 import uuid
 
 from google.cloud import bigtable
@@ -23,6 +22,7 @@ from google.cloud.bigtable import enums
 from mock import Mock, patch
 
 import pytest
+from test_utils.retry import RetryInstanceState
 
 from metricscaler import get_cpu_load
 from metricscaler import get_storage_utilization
@@ -109,32 +109,31 @@ def test_scale_bigtable(instance):
     instance.reload()
 
     cluster = instance.cluster(BIGTABLE_INSTANCE)
-    cluster.reload()
+
+    _nonzero_node_count = RetryInstanceState(
+        instance_predicate=lambda c: c.serve_nodes > 0,
+        max_tries=10,
+    )
+    _nonzero_node_count(cluster.reload)()
+
     original_node_count = cluster.serve_nodes
 
     scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, True)
 
-    for n in range(10):
-        time.sleep(10)
-        cluster.reload()
-        new_node_count = cluster.serve_nodes
-        try:
-            assert (new_node_count == (original_node_count + SIZE_CHANGE_STEP))
-        except AssertionError:
-            if n == 9:
-                raise
+    expected_count = original_node_count + SIZE_CHANGE_STEP
+    _scaled_node_count = RetryInstanceState(
+        instance_predicate=lambda c: c.serve_nodes == expected_count,
+        max_tries=10,
+    )
+    _scaled_node_count(cluster.reload)()
 
     scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, False)
 
-    for n in range(10):
-        time.sleep(10)
-        cluster.reload()
-        final_node_count = cluster.serve_nodes
-        try:
-            assert final_node_count == original_node_count
-        except AssertionError:
-            if n == 9:
-                raise
+    _restored_node_count = RetryInstanceState(
+        instance_predicate=lambda c: c.serve_nodes == original_node_count,
+        max_tries=10,
+    )
+    _restored_node_count(cluster.reload)()
 
 
 def test_handle_dev_instance(capsys, dev_instance):

--- a/samples/metricscaler/requirements-test.txt
+++ b/samples/metricscaler/requirements-test.txt
@@ -1,2 +1,3 @@
 pytest==5.3.2
 mock==3.0.5
+google-cloud-testutils

